packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -0,0 +1,149 @@
+
+/**
+ * Properties for invoking a Lambda function with LambdaInvoke
+ */
+export interface LambdaInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * Lambda function to invoke
+   */
+  readonly lambdaFunction: lambda.IFunction;
+
+  /**
+   * The JSON that will be supplied as input to the Lambda function
+   *
+   * @default - The state input (JSON path '$')
+   */
+  readonly payload?: sfn.TaskInput;
+
+  /**
+   * Invocation type of the Lambda function
+   *
+   * @default InvocationType.REQUEST_RESPONSE
+   */
+  readonly invocationType?: LambdaInvocationType;
+
+  /**
+   * Client context to pass to the function
+   *
+   * @default - No context
+   */
+  readonly clientContext?: string;
+
+  /**
+   * Version or alias of the function to be invoked
+   *
+   * @default - No qualifier
+   */
+  readonly qualifier?: string;
+}
+
+/**
+ * Invoke a Lambda function as a Task
+ *
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-lambda.html
+ */
+export class LambdaInvoke extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private readonly metricsConfig: sfn.TaskMetricsConfig;
+  private readonly policies: iam.PolicyStatement[];
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: LambdaInvokeProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, LambdaInvoke.SUPPORTED_INTEGRATION_PATTERNS);
+
+    // TODO - why not just make the task token an input to the task?
+    // or better yet just default the task token to  $$.Task.Token
+    // that way we can insert it into "payload" on behalf of the user
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN
+      && !sfn.FieldUtils.containsTaskToken(props.payload)) {
+      throw new Error('Task Token is required in `payload` for callback. Use Context.taskToken to set the token.');
+    }
+
+    this.metricsConfig = {
+      metricPrefixSingular: 'LambdaFunction',
+      metricPrefixPlural: 'LambdaFunctions',
+      metricDimensions: { LambdaFunctionArn: this.props.lambdaFunction.functionArn },
+    };
+
+    this.policies = [
+      new iam.PolicyStatement({
+        resources: [this.props.lambdaFunction.functionArn],
+        actions: ['lambda:InvokeFunction'],
+      }),
+    ];
+  }
+
+  /**
+   * Provides the service integration task configuration
+   */
+  protected renderTask(): any {
+    const taskStateConfig: TaskStateConfig = {
+      resourceArn: getResourceArn('lambda', 'invoke', this.integrationPattern),
+      parameters: {
+        FunctionName: this.props.lambdaFunction.functionName,
```

Comment:
Use `functionArn` and this will automatically support versions and aliases.

Reply:
done - I guess we were precluding support for versions/aliases in the current `RunLambdaTask`